sources/osgPlugins/ReaderWriterGLTF.js

```diff
@@ -124,218 +121,29 @@ GLTFLoader.prototype = {
-
-                    glTFFileOrUrl = urlOrFiles[ i ];
-                    break;
-
-                }
-
-            }
-
-        }
-
-        if ( !glTFFileOrUrl ) {
-
-            Notify.warn( 'You did not provided any glTF file' );
-            return Promise.resolve( null );
-
-        }
-
-        return this.loadFile( glTFFileOrUrl, 'gltf' );
-
-    },
-
-    requestFileFromReader: function ( file, fileType ) {
-
-        var defer = Promise.defer();
-
-        if ( !( file instanceof File ) )
-            return defer.reject();
-
-        var reader = new window.FileReader();
-        reader.onload = function ( data ) {
-
-            if ( fileType !== 'blob' )
-                defer.resolve( data.target.result );
-            else {
-                var img = new window.Image();
-                img.src = data.target.result;
-
-                defer.resolve( img );
-            }
-
-        };
-
-        if ( fileType === 'arraybuffer' )
-            reader.readAsArrayBuffer( file );
-        else if ( fileType === 'text' )
-            reader.readAsText( file );
-        else
-            reader.readAsDataURL( file );
-
-        return defer.promise;
-    },
-
-    loadFile: function ( urlOrFile, uri ) {
-
-        if ( this._cachepromise[ uri ] )
-            return this._cachepromise[ uri ];
 
-        if ( !urlOrFile ) {
+    loadFile: Promise.method( function ( uri ) {
 
-            Notify.warn( '\'' + uri + '\' file not found' );
-            return Promise.resolve( null );
-
-        }
-
-        var fileType = this.getFileType( urlOrFile );
-
-        if ( typeof ( urlOrFile ) === 'string' ) {
-
-            // Checks whether the url is relative or absolute
-            if ( uri !== 'gltf' ) {
-
-                if ( urlOrFile.indexOf( 'http://' ) !== 0 && urlOrFile.indexOf( 'https://' ) !== 0 ||
-                    urlOrFile.indexOf( 'www.' ) !== 0 ) {
-
-                    urlOrFile = this._localPath + uri;
-
-                }
-            }
-
-            // Checks whether the url poins toward an image
-            if ( fileType === 'blob' ) {
-
-                var imagePromise = this._inputImgReader.readImageURL( urlOrFile, {
-                    imageLoadingUsePromise: true
-                } );
-                this._cachepromise[ uri ] = imagePromise;
-
-                return imagePromise;
-
-            }
+        if ( this._filesMap.has( uri ) )
+            return this._filesMap.get( uri );
+        var fileType = this.getFileType( uri );
 
-            var promise = requestFile( urlOrFile, {
-                responseType: fileType
-            } ).then( function ( data ) {
-
-                return data;
-
-            } );
-            this._cachepromise[ uri ] = promise;
-
-            return promise;
-
-        }
-
-        var promiseFile = this.requestFileFromReader( urlOrFile, fileType );
+        var promiseFile = this.requestFileFromReader( uri, fileType );
```

Comment:
Oups, bug. But nope it should use the requestFile method ( https://github.com/cedricpinson/osgjs/pull/722/files#diff-29f7b843057ce77ec71c63808ab0c91bR57 ). 

The fileHelper is to read a list of files, like you have when drag'n dropping.